src/xenia/base/math.h

```diff
@@ -59,6 +60,36 @@ T next_pow2(T value) {
   return value;
 }
 
+#if __cplusplus >= 201703L
```

Comment:
This is cool and all, but does it actually work? Last time I checked `__cplusplus` was not really meaningful, especially in Visual Studio.

Reply:
Two possible solutions:

1.  Use [`/Zc:__cplusplus`](https://docs.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=vs-2019) compiler flag for msvc.
2. Handle this in `platform.h` with something like:
```
#if XE_COMPILER_MSVC
#define XE_CPP_STD_VER _MSVC_LANG
#else
#define XE_CPP_STD_VER __cplusplus
#endif
```

See [this link](https://gist.github.com/ax3l/53db9fa8a4f4c21ecc5c4100c0d93c94) for an overview of compiler behavior.
[MSVC predefined macros](https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=vs-2019)

Reply:
I like the `/Zc` variation very much. However, with C++17 we finally have standard feature test macros: https://en.cppreference.com/w/User:D41D8CD98F/feature_testing_macros
I wonder if they're respected by MSVC and other compilers though?

Reply:
Ok, so ditch both approaches, and use `__cpp_lib_gcd_lcm` macro.
This [macro is known](https://godbolt.org/z/vYpEfH) by Visual Studio 2017 version 15.9, gcc 7.3 and clang 5.0 or newer and is enabled when the c++17 standard is active.

Reply:
Excellent, good to know compilers have all adapted it.